sound/soc/sof/loader.c

```diff
@@ -358,6 +360,75 @@ static int snd_sof_fw_ext_man_parse(struct snd_sof_dev *sdev,
 	return ext_man_size;
 }
 
+static int snd_sof_fw_ext_man4_parse(struct snd_sof_dev *sdev, const struct firmware *fw)
+{
+	struct sof_ipc4_fw_modules *module_entry;
+	struct sof_man4_fw_binary_header *fw_header;
+	struct sof_ext_manifest4_hdr *hdr;
+	struct sof_man4_module_config *fm_config;
+	struct sof_man4_module_entry *fm_entry;
+	int i;
+
+	if (fw->size < sizeof(hdr)) {
+		dev_err(sdev->dev, "Invalid fw size\n");
+		return -EINVAL;
+	}
+
+	hdr = (struct sof_ext_manifest4_hdr *)fw->data;
+
+	fw_header = (struct sof_man4_fw_binary_header *)(fw->data + hdr->len +
+							 SOF_MAN4_FW_HDR_OFFSET);
+	dev_dbg(sdev->dev, " fw %s: header length %x, module num %d", fw_header->name,
+		fw_header->len, fw_header->num_module_entries);
+
+	fm_entry = (struct sof_man4_module_entry *)((void *)fw_header + fw_header->len);
+	fm_config = (struct sof_man4_module_config *)(fm_entry + fw_header->num_module_entries);
```

Comment:
if I look at the breakdown of the manifest above, after all the modules you have "FW Content". It doesnt seem to se show module config. Am I reading this wrong?

Reply:
@ranj063 module config is used at line 402 `module_entry->bss_size = fm_config[fm_entry->cfg_offset].is_bytes;`. Does it answer your question?

Reply:
No @bardliao  what I was asking was that on line 45 above, after module [n] , it shows fw content. But in the line above fw_config Is pointing to that isn't i